### DISCUSSION TITLE
TupleTransformer and ResultTransformer

### DISCUSSION THREAD

**Discussion by gavinking** [hibernate/hibernate-orm#4630] [Design Proposals]
H6 introduces `TupleTransformer` and `ResultTransformer` as a replacement for `ResultTransformer`.

I understand that there is pressure from some quarters to have "something" to replace `ResultTransformer`, even though I personally don't see the need.

But I don't think I understand why we need to introduce new interfaces instead of just using `BiFunction<Object[],String[],T>` and `Function<List<Object>,List<T>>` in the signature of `Query.set TupleTransformer()` and `Query.set ResultTransformer()`. To me, those are the built-in types that Java provides for this, and we don't need to define anything new.

WDYT, @sebersole ?

**Comment by sebersole** [comment]
The new interfaces are Functional.  I just prefer the actual contract, users can certainly pass in lambdas, etc

**Reply by gavinking** [reply]
OK, fine, it's NBD.
